### PR TITLE
Feature/antra login component

### PR DESCRIPTION
**Pull request by narendarlgentyala-antra**
Hi Patrick, 

I have designed the Login Component and Written Some Testcases.

Kindly have a look and let me know if it is good to merge with the master branch.

### CODE REVIEW

File: projects/antra-ui/src/lib/components/login/login.component.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  /**
+   * @ignore
+   */
+  ngOnInit(): void {}
+
+  /**
+   * @ignore
+   */
+  // tslint:disable-next-line: typedef
+  sendActionName(actionName: string) {
```
Comment (Show3567) [inline projects/antra-ui/src/lib/components/login/login.component.ts]: 1. Maybe we need a type named like loginActions and list these strings ('OFFICE365', 'GMAIL' ...) to limited developer input, and also create documents for these emitted strings to let the developer know what to do next after getting these strings.

File: projects/antra-ui/src/lib/components/login/login.component.html [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      </span>
+      <span class="text-danger" *ngIf="password.touched && password.errors?.invalidPassword">
+        {{ passwordValidationMessage[1] }}
+      </span>
+    </div>
+    <div class="form-group">
+      <button type="submit" class="btn btn-primary">LOGIN</button>
+    </div>
+    <div align="right" class="margin-top">
+      <b class="light-color"><span (click)="sendActionName('FORGOTPASSWORD')">Forgot Password</span> </b>
```
Comment (Show3567) [inline projects/antra-ui/src/lib/components/login/login.component.html]: 1. Move the mouse on the "Forgot Password" may change the curser as a pointer or just make it a link. Now, it seems like a normal text on the login dialog.
image

File: projects/antra-ui/src/lib/components/login/login.component.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+})
+/**
+ * Example of usage:
+ * <example-url>http://127.0.0.1:4200/components/login</example-url>
+ */
+export class LoginComponent implements OnInit {
+  @Output() clickEvent = new EventEmitter<string>();
+
+  loginModel = new Login();
+  emailAddressValidationMessage = ['Email Address is required', 'Please Enter Valid Email Address'];
```
Comment (Show3567) [inline projects/antra-ui/src/lib/components/login/login.component.ts]: 1. Maybe the validated message can be customized by the developer, and also the validated logic can be customized by the developer.

File: projects/antra-ui/src/lib/components/login/login.component.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  passwordValidationMessage = [
```
Comment (Show3567) [inline projects/antra-ui/src/lib/components/login/login.component.ts]: 1. Maybe the validated message can be customized by the developer, and also the validated login can be customized by the developer.

File: projects/antra-ui/src/lib/components/login/login.component.html [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        </td>
+        <td width="50%" align="right">
+          <button type="button" (click)="sendActionName('OFFICE365')" class="btn btn-warning">
+            <img align="left" class="btn-img" src="../assets/svg/office.svg" />Office 365
+          </button>
+        </td>
+      </tr>
+    </table>
+  </form>
+</div>
```
Comment (Show3567) [inline projects/antra-ui/src/lib/components/login/login.component.html:65]: 1. Aftershow these valid messages, "Login Using" buttons be pushed out of the login dialog.
image

File: projects/antra-ui/src/lib/components/login/login.component.html [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    </div>
+    <hr class="light-color" />
+    <div class="margin-top">
+      <b class="light-color" style="font-weight: bold">Login using</b>
+    </div>
+
+    <table width="100%">
+      <tr>
+        <td width="50%">
+          <button class="btn btn-danger" type="button" (click)="sendActionName('GMAIL')">
```
Comment (Show3567) [inline projects/antra-ui/src/lib/components/login/login.component.html]: 1. Sometimes, some "Login Using" button can be disabled.

File: projects/antra-ui/src/lib/components/login/login.component.html [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    </div>
+    <div class="form-group">
+      <button type="submit" class="btn btn-primary">LOGIN</button>
+    </div>
+    <div align="right" class="margin-top">
+      <b class="light-color"><span (click)="sendActionName('FORGOTPASSWORD')">Forgot Password</span> </b>
+    </div>
+    <hr class="light-color" />
+    <div class="margin-top">
+      <b class="light-color" style="font-weight: bold">Login using</b>
```
Comment (Show3567) [inline projects/antra-ui/src/lib/components/login/login.component.html]: 1. Avoid using inline-style.

File: projects/antra-ui/src/lib/components/login/login.component.html [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    </div>
+
+    <table width="100%">
```
Comment (Show3567) [inline projects/antra-ui/src/lib/components/login/login.component.html]: Avoid using inline-style.

File: projects/antra-ui/src/lib/components/login/login.component.html [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+      <tr>
+        <td width="50%">
```
Comment (Show3567) [inline projects/antra-ui/src/lib/components/login/login.component.html]: Avoid using inline-style.

File: projects/antra-ui/src/lib/components/login/login.component.html [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+          <button class="btn btn-danger" type="button" (click)="sendActionName('GMAIL')">
+            <img align="left" class="btn-img" src="../assets/svg/gmail.svg" />Gmail
+          </button>
+        </td>
+        <td width="50%" align="right">
```
Comment (Show3567) [inline projects/antra-ui/src/lib/components/login/login.component.html]: Avoid using inline-style.

File: projects/antra-ui/src/lib/components/login/login.component.html (hunk 6 above)
Comment (narendarlgentyala-antra) [reply projects/antra-ui/src/lib/components/login/login.component.html]: Hi David,

Could you please elaborate on the above issue?

Comment (Show3567) [reply projects/antra-ui/src/lib/components/login/login.component.html]: Sure, I think sometimes the requirement may need to disable these buttons inside the "Login Using" table.

File: projects/antra-ui/src/lib/components/login/login.component.html (hunk 2 above)
Comment (narendarlgentyala-antra) [reply projects/antra-ui/src/lib/components/login/login.component.html]: Changes Done. Thanks.

File: projects/antra-ui/src/lib/components/login/login.component.html (hunk 5 above)
Comment (narendarlgentyala-antra) [reply projects/antra-ui/src/lib/components/login/login.component.html:65]: Changes Done. Thanks.

File: projects/antra-ui/src/lib/components/login/login.component.html (hunk 10 above)
Comment (narendarlgentyala-antra) [reply projects/antra-ui/src/lib/components/login/login.component.html]: Changes Done. Thanks.

File: projects/antra-ui/src/lib/components/login/login.component.html (hunk 9 above)
Comment (narendarlgentyala-antra) [reply projects/antra-ui/src/lib/components/login/login.component.html]: Changes Done. Thanks.

File: projects/antra-ui/src/lib/components/login/login.component.html (hunk 8 above)
Comment (narendarlgentyala-antra) [reply projects/antra-ui/src/lib/components/login/login.component.html]: Changes Done. Thanks.

File: projects/antra-ui/src/lib/components/login/login.component.html (hunk 7 above)
Comment (narendarlgentyala-antra) [reply projects/antra-ui/src/lib/components/login/login.component.html]: Changes Done. Thanks.

File: projects/antra-ui/src/lib/components/login/login.component.ts (hunk 1 above)
Comment (narendarlgentyala-antra) [reply projects/antra-ui/src/lib/components/login/login.component.ts]: Done. Thanks

File: projects/antra-ui/src/lib/components/login/login.component.ts (hunk 4 above)
Comment (narendarlgentyala-antra) [reply projects/antra-ui/src/lib/components/login/login.component.ts]: Changes Done. Thanks

File: projects/antra-ui/src/lib/components/login/login.component.ts (hunk 3 above)
Comment (narendarlgentyala-antra) [reply projects/antra-ui/src/lib/components/login/login.component.ts]: Changes Done. Thanks

File: projects/antra-ui/src/lib/components/login/login.component.ts (hunk 4 above)
Comment (Show3567) [reply projects/antra-ui/src/lib/components/login/login.component.ts]: Sorry for the typo, the logic may be customized by the developer.

**Pull request merged**